### ISSUE TITLE
inspec.yml and inputs-file interpret hashes differently

### ISSUE THREAD

**Issue by samcornwell** [inspec/inspec#5156]
Using Inspec 4.21.1. Whenever I define an `input` in `inspec.yml` which is a hash, it gets interpreted differently than if I pass it as an input file. The `inspec.yml` turns the keys into symbols, while and input file turns the keys into strings.

Example:
inspec.yml
```
inputs:
  - name: sns_subscriptions
      description: 'SNS subscription list and details'
      type: hash
      value:
        subscription_arn1:
          endpoint: endpoint_value
          owner: owner_value
          protocol: protocol_value
        subscription_arn2:
          endpoint: endpoint_value
          owner: owner_value
          protocol: protocol_value
```

output of `input('sns_subscriptions')`
```
{:subscription_arn1=>{:endpoint=>“endpoint_value”, :owner=>“owner_value”, :protocol=>“protocol_value”}, :subscription_arn2=>{:endpoint=>“endpoint_value”, :owner=>“owner_value”, :protocol=>“protocol_value”}}
```

Example:
Override with an `inputs.yml` (inputs file)
```
sns_subscriptions:
  subscription_arn1:
    endpoint: endpoint_value
    owner: owner_value
    protocol: protocol_value
  subscription_arn2:
    endpoint: endpoint_value
    owner: owner_value
    protocol: protocol_value
```

output of `input('sns_subscriptions')`
```
{"subscription_arn1"=>{"endpoint"=>"endpoint_value", "owner"=>"owner_value", "protocol"=>"protocol_value"}, "subscription_arn2"=>{"endpoint"=>"endpoint_value", "owner"=>"owner_value", "protocol"=>"protocol_value"}}
```

Aha! Link: https://chef.aha.io/features/SH-2650

**Comment by samcornwell** [comment]
For what it's worth, I think it makes the most sense for both methods to produce strings for keys, rather than Ruby-specific symbols.

**Comment by clintoncwolfe** [comment]
This was fixed by #5446
